agent/lib/queue.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var create_podgroup = require('./podgroup.js');
+var kubernetes = require('../lib/kubernetes.js');
+var log = require('./log.js').logger();
+var pod_watcher = require('./pod_watcher');
+
+function Queue (name) {
+    this.name = name;
+    this.pods  = create_podgroup();
+};
+
+function get_current_pods () {
+
+    var options = {
+        selector: 'role=broker'
+    };
+
+    if (process.env.INFRA_UUID) {
+        var infraSelector = "infraUuid=" + process.env.INFRA_UUID;
+        if (options.selector) {
+            options.selector += "," + infraSelector;
+        }
+    }
+    return kubernetes.get('pods', options);
+}
+
+function find_queue (name, pods) {
+    return Promise.all(pods.map(function (pod) {
+        return pod.broker.getQueueNames();
+    } )).then(function (results) {
+            var found = false;
+            var pod = undefined;
+            var min = undefined;
+            for (var i = 0; !found && i < results.length; i++) {
+                if (results[i].indexOf(name) >= 0) {
+                    pod = pods[i];
+                    found = true;
```

Comment:
You are indexing the pod array (pods) with an index that relates to the queue array.  That seems wrong.

Reply:
Relates to the comment above.
'i' is the index of the results for all the promises.  eg. [ ['foo', 'bar'], ['queue_on_other_pod'] ]
The 1st pod queried returned 'foo', and 'bar', and the 2nd pod queried returned 'queue_on_other_pod'.
The results are in the same order as the promises in Promise.all